src/database/UIDatabase.js

```diff
@@ -7,18 +7,27 @@ export class UIDatabase {
   objects(type) {
     const results = this.database.objects(translateToCoreDatabaseType(type));
     switch (type) {
+      case 'CustomerInvoice':
+        // Only show invoices generated from requisitions once finalised
+        return results.filtered('type == "customer_invoice"'
+                                + ' AND (linkedRequisition == null OR status == "finalised")');
+      case 'SupplierInvoice':
+        return results.filtered('type == "supplier_invoice"'
+                                + ' AND otherParty.type != "inventory_adjustment"');
       case 'Customer':
         return results.filtered('isVisible == true AND isCustomer == true');
       case 'Supplier':
         return results.filtered('isVisible == true AND isSupplier == true');
       case 'InternalSupplier':
         return results.filtered('isVisible == true AND isSupplier == true AND type == "store"');
       case 'ExternalSupplier':
-        return results.filtered(
-          "isVisible == true AND isSupplier == true AND type == 'facility'"
-        );
+        return results.filtered('isVisible == true AND isSupplier == true AND type == "facility"');
       case 'Item':
         return results.filtered('isVisible == true');
+      case 'RequestRequisition':
+        return results.filtered('type == "request"');
+      case 'ResponseRequisition':
+        return results.filtered('serialNumber != "-1" AND type == "response"');
```

Comment:
Breaks post sync processor again I believe. This is the right way to do this. I'll commit my suggested fix in a moment.